src/utils/indexer/getBlockTime.ts

```diff
@@ -1,49 +1,26 @@
-import { getSdk } from "../cache/getSdk";
-import { logger } from "../logger";
+import { eth_getBlockByNumber, getRpcClient } from "thirdweb";
+import { getChain } from "../chain";
+import { thirdwebClient } from "../sdk";
 
-const KNOWN_BLOCKTIME_SECONDS = {
-  1: 12,
-  137: 2,
-} as Record<number, number>;
+export const getBlockTimeSeconds = async (
+  chainId: number,
+  blocksToEstimate: number,
+) => {
+  const chain = await getChain(chainId);
+  const rpcRequest = getRpcClient({
+    client: thirdwebClient,
+    chain,
+  });
 
-const DEFAULT_BLOCKTIME_SECONDS = 10;
-const BLOCKS_TO_ESTIMATE = 100;
+  const latestBlock = await eth_getBlockByNumber(rpcRequest, {
+    blockTag: "latest",
+    includeTransactions: false,
+  });
+  const referenceBlock = await eth_getBlockByNumber(rpcRequest, {
+    blockNumber: latestBlock.number - BigInt(blocksToEstimate),
+    includeTransactions: false,
+  });
 
-export const getBlockTimeSeconds = async (chainId: number) => {
```

Comment:
The previous logic would add up all timestamps between the last 100 blocks. This should be the same as just taking the diff the current block and the n-100 block.